### PR TITLE
drivers: entropy: Fix return of error code, style and naming

### PR DESCRIPTION
**Pull request by joerchan**
Fix cc3xx entropy driver error code handling:
- Use psa_status_t type for PSA APIs.
- Don't return errors from nrf_cc3xx APIs which are not errno values.
- Don't initialize variables when the assigned value will not be used.

Fix indentation and style in the cc3xx entropy driver according to the
coding standard.

Rename the source file for the cc3xx entropy driver to match its
kconfig name.

### CODE REVIEW

**Review by Vge0rge — APPROVED** [review]
LGTM!

**Comment by joerchan** [comment]
@anangl Will you be able to have a look?

File: drivers/entropy/entropy_cc3xx.c [hunk 1]
```diff
@@ -29,16 +29,19 @@
...
 
 #if defined(CONFIG_BUILD_WITH_TFM)
-	res = psa_generate_random(buffer, length);
-	if (res != PSA_SUCCESS) {
-		return -EINVAL;
+	psa_status_t status;
+
+	status = psa_generate_random(buffer, length);
+	if (status != PSA_SUCCESS) {
+		err = 0;
```
Comment (anangl) [inline drivers/entropy/entropy_cc3xx.c:41]: Shouldn't `0` be returned when `status == PSA_SUCCESS`?

IMHO this `err` variable does not really help, so I'd suggest to get rid of it and simply return `-EINVAL` here and `0` below. And similarly for the non-TFM path - return an error when something goes wrong and `0` at the end.

File: drivers/entropy/entropy_cc3xx.c [hunk 2]
```diff
@@ -26,98 +26,99 @@
...
 	}
-#endif
 
-	return res;
+	if (offset == length) {
+		err = 0;
+	}
+
+	return err;
+#endif
```
Comment (anangl) [inline drivers/entropy/entropy_cc3xx.c:97]:
```suggestion
#endif /* defined(CONFIG_BUILD_WITH_TFM) */
```

File: drivers/entropy/entropy_cc3xx.c (hunk 1 above)
Comment (joerchan) [reply drivers/entropy/entropy_cc3xx.c]: Thanks. Fixed.

**Review by anangl — APPROVED** [review]

**Pull request merged**